cpp/src/arrow/type.h

```diff
@@ -288,31 +288,46 @@ std::shared_ptr<DataType> GetPhysicalType(const std::shared_ptr<DataType>& type)
 class ARROW_EXPORT FixedWidthType : public DataType {
  public:
   using DataType::DataType;
+  // This is only for preventing defining this class in each
+  // translation unit to avoid one-definition-rule violation.
+  ~FixedWidthType() override;
```

Comment:
You probably don't need to define the destructor separately?
```suggestion
  ~FixedWidthType() override = default;
```

Reply:
I tried `= default`. If we add `= default`, generated symbol is still `W` not `U`. It seems that we need to define the destructor separately for `U`.